README.md

```diff
@@ -1 +1,218 @@
+
+
+```js
+emitter.on('knock', function() {
+    console.log('Who\'s there?')
+})
+
+emitter.on('knock', function() {
+    console.log('Go away!')
+})
+```
+
+```js
+emitter.emit('knock')
+```
+
+Run your newly created Node script with this command:
+
+```
+node knock-knock.js
+```
+
+You should see the output produced by two event listeners in the order in which they were defined because the event `knock` was triggered by the last statement:
+
+```
+Who's there?
+Go away!
+```
+
+Just for fun, add another `emitter.emit('knock')`. There would be there output:
+
+```
+Who's there?
+Go away!
+Who's there?
+Go away!
+```
+
+To summarize, we defined two different event listeners for the same arbitrary event `knock`. Then we triggered the event twice. 
+
+Additional methods include:
+
+* `emitter.listeners(eventName)`: List the event listeners associated with the `eventName` event
+* `emitter.once(eventName, listener)`: Create event listener for `eventName` and execute it just one time. This is similar to `.on()` but fires only once.
+* `emitter.removeListener(eventName, listener)`: Remove event listener for `eventName`
+
+In our knock-knock example, we used the observer object `event`. It's not very interesting because it didn't do much. What's more interesting and useful is that we can extend Event Emitter properties and methods into ANY OBJECT/class. There's this convenient function `inherits()` from the core `util` module. It takes a child (`fancyChildObjectWithEvents`) and parent objects, and makes the child inherit from the parent (Event Emitter).
+
+```js
+var util = require('util')
+var events  = require('events')
+util.inherits(fancyChildObjectWithEvents, events.EventEmitter)
+```
+
+Imagine a situation when you're developing an email job module. This module needs to be flexible enough to send different notifications when the job is done: weekly, daily and monthly. 
+
+Event listeners can be located in different places than the class. This feature is useful for our email job class. We will create the class in `job.js`, then set up event listeners in `weekly.js`.
+
+The `job.js` has module imports:
+
+```js
+// job.js
+var util = require('util')
+var events = require('events')
+```
+
+Then create a class/object `Job` using pseudo-classical pattern. It must have `process` method which will launch the email job. Inside of that method, emit the `done` event when everything is finished - async! And one more thing, we can pass any data in the `emit` method. In this example, pass the time&date.
+
+```js
+var Job = function Job() {
+  var job = this // We'll use it later to pass 'this' to closures
+  // ...
+  job.process = function() {
+    // ...
+    setTimeout(function(){ // Emulate the delay of the job - async!
+      job.emit('done', { completedOn: new Date() })
+    }, 700)
+  }
+}
+```
+
+
+
+Now, extend the events and export the class:
+
+```js
+util.inherits(Job, events.EventEmitter)
```

Comment:
I like your example for its clarity, but I would love to see something that users can dig into. Could we have them code along with a command line prompt that they write?